osquery/tables/applications/posix/docker.cpp

```diff
@@ -578,26 +589,42 @@ QueryData genContainerProcesses(QueryContext& context) {
         std::vector<std::string> vector;
         BOOST_FOREACH (const auto& v, processes.second) {
           vector.push_back(v.second.data());
-        }
-
-        // On OS X, ps_args are not being honored. Check number of columns
-        if (vector.size() != 21) {
-          continue;
+          VLOG(1) << v.second.data() << "\n";
         }
 
         Row r;
         r["id"] = id;
         r["pid"] = BIGINT(vector.at(0));
-        r["name"] = vector.at(19);
-        r["cmdline"] = vector.at(20);
+        r["wired_size"] = BIGINT(0); // No support for unpagable counters
+#if defined(__APPLE__)
+        r["state"] = "";
+        r["uid"] = BIGINT(vector.at(1));
+        r["gid"] = -1;
+        r["euid"] = -1;
+        r["egid"] = -1;
+        r["suid"] = -1;
+        r["sgid"] = -1;
+        r["resident_size"] = -1;
+        r["total_size"] = -1;
+        r["start_time"] = -1;
+        r["parent"] = -1;
+        r["pgroup"] = -1;
+        r["threads"] = -1;
+        r["nice"] = -1;
+        r["user"] = "";
+        r["time"] = vector.at(2);
+        r["cpu"] = -1;
+        r["mem"] = -1;
+        r["name"] = "";
+        r["cmdline"] = vector.at(3);
+#elif defined(__linux__)
         r["state"] = vector.at(1);
         r["uid"] = BIGINT(vector.at(2));
         r["gid"] = BIGINT(vector.at(3));
         r["euid"] = BIGINT(vector.at(4));
         r["egid"] = BIGINT(vector.at(5));
         r["suid"] = BIGINT(vector.at(6));
         r["sgid"] = BIGINT(vector.at(7));
-        r["wired_size"] = BIGINT(0); // No support for unpagable counters
```

Comment:
Teddy wanted to keep compatibility with `processes` columns. So this was left in there...